src/main/java/graphql/execution/ExecutionStrategy.java

```diff
@@ -325,7 +326,17 @@ DeferredExecutionSupport createDeferredExecutionSupport(ExecutionContext executi
                 Object fieldValueInfo = resolveFieldWithInfo(executionContext, newParameters);
                 futures.addObject(fieldValueInfo);
             }
+
         }
+
+        if (executionContext.hasIncrementalSupport()
+                && deferredExecutionSupport.deferredFieldsCount() > 0
+                && executionContext.getGraphQLContext().getBoolean(IncrementalExecutionContextKeys.ENABLE_EAGER_DEFER_START, false)) {
+
+            executionContext.getIncrementalCallState().startDeferredCalls();
```

Comment:
This call here is only need as a side effect to create the memoizing supplier (of the Publisher)

I think we should invent a new method that does that and hides the side effect

`executionContext.getIncrementalCallState().startEarlyDraining();`